python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -193,70 +204,53 @@ async def write(self, data: str) -> None:
         assert self.client is not None  # make mypy happy
         await self.client.write(data.encode() + b"\r\n")
 
-    async def read_loop(self) -> None:
+    async def setup_reading(self) -> None:
+        self._have_seen_data = False
+        if self.connected and self.simulation_mode == 0:
+            # Make sure that the correct frequency range is used by the
+            # spectrum analyzer.
+            await self.write(self.set_freq_start_cmd)
+            await self.write(self.set_freq_stop_cmd)
+
+    async def read_data(self) -> None:
         """Read raw data from the SSA3000X Spectrum Analyzer."""
+        timestamp = utils.current_tai()
```

Comment:
If the connection is lost, is the resulting exception clear? If not, please consider checking and raising at the start of this method.